### PR TITLE
Like post

### PR DESCRIPTION
**Pull request by manish591**
### Changes
* Like post functionality
* Dislike post functionality

### CODE REVIEW

File: src/components/user-post/UserPost.jsx [hunk 1]
```diff
@@ -89,26 +96,44 @@ const UserPost = ({ post, setIsOpenPostModal }) => {
...
+                      }}>
+                      <span className="material-icons-outlined text-xl">
+                        favorite
+                      </span>
+                    </button>
+                  ) : (
+                    <button
+                      type="button"
+                      onClick={() => {
+                        dispatch(likePost({ postId: _id, token }));
```
Comment (praveenreddy1798) [inline src/components/user-post/UserPost.jsx:117]: Do we have an handler to handle if the user is not logged in and he tries to like the post. We need to redirect the user to the login page in that case. Please add an handler if you don't have one

**Review by praveenreddy1798 — APPROVED** [review]

File: src/components/user-post/UserPost.jsx (hunk 1 above)
Comment (manish591) [reply src/components/user-post/UserPost.jsx:117]: User is allowed to like only on logged in. The user feed route is a protected route so unauthorised users can't access the feed thus, they can't like the post.

**Review by maitrakhatri — APPROVED** [review]
Suggestion: Add a "use guest credentials" button to quickly login and test the implemented features. Also the redux actions are getting console logged, so take care of them, remove if not needed. 
image

Clean code, easy to read and understand.

**Comment by manish591** [comment]
> Suggestion: Add a "use guest credentials" button to quickly login and test the implemented features. Also the redux actions are getting console logged, so take care of them, remove if not needed. image
> 
> Clean code, easy to read and understand.

I'm using redux logger middleware that's why it is showing in the console. Will remove it.

**Pull request merged**